packages/pyright-internal/src/common/realFileSystem.ts

```diff
@@ -46,7 +47,10 @@ function getArchivePart(path: string) {
     if (idx <= 0) {
         idx = path.indexOf(DOT_EGG);
         if (idx <= 0) {
-            return null;
+            idx = path.indexOf(DOT_JAR);

```

Comment:
This technique for determining the file extension is not very robust. It doesn't properly handle file names with multiple dots. It also doesn't handle names like "x.zipx" or "a.jarxxxxx".

There is a function in pathUtils called `getFileExtension` that handles proper extraction of file extension. You can also use `stripFileExtension` if you want to strip the extension.

Reply:
This function also supports paths within a zip file, treating the zip like a directory. For example: `c:\dir\file.zip\module.py`.

The second-to-last line verifies that the extension is either at the end of the path or is immediately followed by the path separator char.

So I believe it is properly handling the scenarios you mentioned and `getFileExtension` won't help here.

Reply:
@erictraut, based on my comment above are you ok with this change?

Reply:
Yep